eds/blocks/hero/hero.js

```diff
@@ -1,19 +1,53 @@
-import { domEl } from '../../scripts/dom-helpers.js';
-
 export default function decorate(block) {
-  block.querySelectorAll('picture > img').forEach((img, i) => {
-    img.setAttribute('loading', 'eager');
-    if (i === 0) {
-      img.closest('p').classList.add('foreground-img');
+  // Find all children of the block
+  const blockChildren = [...block.children];
+
+  // Iterate over each child of the block
+  // Remove the first child
+  // Return the second child
+  const secondChildren = blockChildren.map((child) => {
+    if (child.children.length > 0) {
+      child.removeChild(child.children[0]);
+    }
+    return child.children[0];
+  });
+
+  secondChildren.forEach((child) => {
+    if (child) {
+      const parentDiv = child.parentElement;
+      parentDiv.replaceWith(child);
     }
   });
+
+  // For each secondChildren, set the first to hero-content,
+  // the second to hero-image, third to hero-video,
+  // last to hero-background
+  secondChildren.forEach((child, i) => {
+    if (child) {
+      if (i === 0) {
+        child.classList.add('hero-content');
+      } else if (i === 1) {
+        child.classList.add('hero-image');
+      } else if (i === 2) {
+        child.classList.add('hero-video');
+      } else {
+        child.classList.add('hero-background');
+      }
```

Comment:
A couple issues here:
- This implementation seems "order-dependent", you should be using the identifiers (content, image, backgroundImage, ...) to identify each of the elements, otherwise this can easily break if a field is missing/in a different order.
- Rather than prefix all these elements with "hero" the best practice is to prefix them with "hero".

Reply:
>>Rather than prefix all these elements with "hero" the best practice is to prefix them with "hero".

I think something got dropped here? 

I'll look at improving the logic. Point taken

Reply:
I meant them to prefix the css selector with .hero/use css nesting (like .hero .content, .hero .video, ...)